### PR TITLE
optimizer: Fix a bug in resolve having identifier

### PR DESCRIPTION
**Pull request by shenli**
Fix sqllogic test failure
@coocood @qiuyesuifeng

### CODE REVIEW

File: optimizer/resolver.go [hunk 1]
```diff
@@ -370,10 +370,11 @@ func (nr *nameResolver) resolveColumnNameInContext(ctx *resolverContext, cn *ast
 				return true
 			}
 		}
-		return nr.resolveColumnInResultFields(cn, ctx.fieldList)
+		found := nr.resolveColumnInResultFields(ctx, cn, ctx.fieldList)
+		return found
```
Comment (qiuyesuifeng) [inline optimizer/resolver.go]: Directly `return nr.resolveColumnInResultFields(ctx, cn, ctx.fieldList)`?

Comment (shenli) [inline optimizer/resolver.go]: PTAL @qiuyesuifeng

**Comment by qiuyesuifeng** [comment]
@coocood  PTAL

**Comment by coocood** [comment]
LGTM

**Pull request merged**